### ISSUE TITLE
Task timeouts, errors, and not working as expected

### ISSUE THREAD

**Issue by nitrocode** [schubergphilis/terraform-aws-mcaf-securityhub-findings-manager#20]
I only have a single rule in my `suppressions.yml` file

```yaml
Suppressions:
  # S3.6 S3 permissions granted to other AWS accounts in bucket policies should be restricted
  # https://docs.aws.amazon.com/securityhub/latest/userguide/s3-controls.html#s3-6
  "S3.6":
    - action: SUPPRESSED
      notes: |
        This is a suppression test
      rules:
        - arn:aws:s3:::my-unique-test-bucket
```

I see a lot of these errors

```
[ERROR] ValueError: Skipping execution because status is None. Valid statuses: ['FAILED', 'HIGH', 'WARNING']
Traceback (most recent call last):
  File "/var/task/aws_lambda_powertools/logging/logger.py", line 442, in decorate
    return lambda_handler(event, context, *args, **kwargs)
  File "/var/task/securityhub_events.py", line 210, in lambda_handler
    validate_event(event)
  File "/var/task/securityhub_events.py", line 173, in validate_event
    raise ValueError(f'Skipping execution because status is {status}. Valid statuses: {VALID_STATUSES}')
```

The dynamodb table seems to get a putitem and then a deleteitem very quickly.

I tried bumping the timeout from 120 to 300 and I see timeout errors at 300 which is odd.

This is the last message I see prior to the timeout.

https://github.com/schubergphilis/terraform-aws-mcaf-securityhub-findings-manager/blob/7027a792334d96beee0db667946a29256e28988a/files/lambda-artifacts/securityhub-suppressor/securityhub_events.py#L65

I also see new logs created every 5 to 12 seconds which seems more than expected. Some of them finish very quickly in 5 to 20 seconds and the rest timeout.

Also my one s3 bucket is not being suppressed and I do not know why.

---

Another example of this is by aws which also uses lambda, event bridge rule, sqs, etc but without notes

https://aws.amazon.com/blogs/security/how-to-create-auto-suppression-rules-in-aws-security-hub/

---

Due to the cost, rather than debugging this set of lambdas, it might be easier to read a similar `suppressions.yml` file and use a python script in github actions to run on a daily cron to suppress these items...

**Comment by carlaabanes** [comment]
was ever fixed? I am having the same set of errors. Never worked for my case, even though I only have one suppression rule in my yml file.

**Comment by marwinbaumannsbp** [comment]
@nitrocode I don't think your example is a valid suppression rule, we have updated the README: https://github.com/schubergphilis/terraform-aws-mcaf-securityhub-findings-manager/tree/v2.0.0#examples can you retest?

**Comment by nitrocode** [comment]
Hi @marwinbaumannsbp thank you for the recent changes in this module. I'm not sure how my yaml config is different from the example other than the omission of the ^ in the beginning and the $ at the end. Both of these should work, no?

```hcl
      rules:
        - arn:aws:s3:::my-unique-test-bucket
```

I can try retesting but I have no way to keep the lambdas from timing out which makes this app very costly.

**Comment by nitrocode** [comment]
Yes. Currently I am not consolidating findings.

![image](https://github.com/schubergphilis/terraform-aws-mcaf-securityhub-findings-manager/assets/7775707/7cc41e33-108f-4edd-916d-7af5ee405667)

Does it need to be consolidated for this app to work?

**Comment by marwinbaumannsbp** [comment]
Hi @nitrocode, it needs to be off indeed, the module does not support consolidated findings.

**Comment by marwinbaumannsbp** [comment]
@nitrocode this module and the rule logic has been rewritten: https://github.com/schubergphilis/terraform-aws-mcaf-securityhub-findings-manager/blob/main/UPGRADING.md#behaviour-v300 we now support consolidated findings and the suppression logic (now calles rules) have been rewritten. Can you try again?

**Comment by nitrocode** [comment]
@marwinbaumannsbp I had consolidated findings turned off when I gave this a try. I ended up giving up because I wasn't sure how to troubleshoot this and opted for something simpler for my needs. 

I created this module https://github.com/infralicious/terraform-aws-securityhub-batchupdatefindings which uses a similar yaml format except that it only works on specific findings rather than specific resources.
